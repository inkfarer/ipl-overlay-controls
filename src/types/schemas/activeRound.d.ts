/* tslint:disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export type ActiveRoundTeam = {
	id: string;
	name: string;
	logoUrl?: string;
	showLogo: boolean;
	players: {
		name: string;
		[k: string]: unknown;
	}[];
	[k: string]: unknown;
} & {
	score: number;
	color: string;
	[k: string]: unknown;
};

export interface ActiveRound {
	teamA: ActiveRoundTeam;
	teamB: ActiveRoundTeam;
	activeColor: {
		index: number;
		title: string;
		categoryName: string;
		isCustom: boolean;
		clrNeutral: string;
	};
	match: {
		id: string;
		name: string;
		type: 'BEST_OF' | 'PLAY_ALL';
		isCompleted: boolean;
	};
	games: {
		winner: 'none' | 'alpha' | 'bravo';
		stage: string;
		mode: string;
		color?: {
			index: number;
			title: string;
			clrA: string;
			clrB: string;
			clrNeutral: string;
			categoryName: string;
			isCustom: boolean;
			[k: string]: unknown;
		} & {
			colorsSwapped: boolean;
			[k: string]: unknown;
		};
		[k: string]: unknown;
	}[];
}
