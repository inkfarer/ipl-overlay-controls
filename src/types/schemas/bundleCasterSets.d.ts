/* prettier-ignore */
/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface BundleCasterSets {
	[k: string]: {
		[k: string]: {
			[k: string]: Caster;
		};
	};
}
export interface Caster {
	name: string;
	twitter: string;
	pronouns: string;
	imageUrl?: string;
	videoUrl?: string;
}
