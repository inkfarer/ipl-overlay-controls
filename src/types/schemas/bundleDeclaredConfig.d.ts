/* prettier-ignore */
/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface BundleDeclaredConfig {
	[k: string]: {
		scenes: {
			value: string;
			names?: {
				EN?: string;
				JA?: string;
				[k: string]: string;
			};
		}[];
	};
}
