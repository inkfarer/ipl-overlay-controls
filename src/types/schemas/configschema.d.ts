/* prettier-ignore */
/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface Configschema {
	screenshotParser?: {
		saveBadScreenshotsToDisk?: boolean;
	};
	lastfm?: {
		apiKey?: string;
		secret?: string;
	};
	smashgg?: {
		apiKey?: string;
	};
	radia?: {
		url?: string;
		socketUrl?: string;
		authentication?: string;
	};
	sendouInk?: {
		apiKey: string;
	};
}
