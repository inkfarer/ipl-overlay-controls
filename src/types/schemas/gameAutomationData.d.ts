/* tslint:disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface GameAutomationData {
	actionInProgress?: 'END_GAME' | 'START_GAME' | 'NONE';
	nextTaskForAction?: {
		name: string;
		index: number;
		executionTimeMillis: number;
	} | null;
}
