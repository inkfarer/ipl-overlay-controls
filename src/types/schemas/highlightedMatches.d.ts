/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export type HighlightedMatches = Match[];

export interface Match {
	meta: {
		name: string;
		/**
		 * Name to be automatically assigned as the name of the next match on graphics
		 */
		shortName: string;
		/**
		 * ID of the match, for Battlefy use _id
		 */
		id: string;
		/**
		 * Name of the stage match originates from
		 */
		stageName?: string;
		/**
		 * The time the set was completed at, null if yet to be completed, Stored in ISO format
		 */
		completionTime?: string;
		/**
		 * Round number from provider
		 */
		round?: number;
		/**
		 * Match number from provider
		 */
		match?: number;
		playType?: 'BEST_OF' | 'PLAY_ALL';
	};
	teamA: Team;
	teamB: Team;
}
export interface Team {
	id: string;
	name: string;
	logoUrl?: string;
	showLogo: boolean;
	players: Player[];
	[k: string]: unknown;
}
export interface Player {
	name: string;
	[k: string]: unknown;
}
