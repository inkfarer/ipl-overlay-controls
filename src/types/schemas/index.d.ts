/* eslint-disable */
// @ts-ignore
export * from './configschema';
// @ts-ignore
export * from './activeBreakScene';
// @ts-ignore
export * from './activeRound';
// @ts-ignore
export * from './assetPaths';
// @ts-ignore
export * from './casters';
// @ts-ignore
export * from './gameAutomationData';
// @ts-ignore
export * from './highlightedMatches';
// @ts-ignore
export * from './lastFmNowPlaying';
// @ts-ignore
export * from './lastFmSettings';
// @ts-ignore
export * from './localeInfo';
// @ts-ignore
export * from './mainFlavorText';
// @ts-ignore
export * from './manualNowPlaying';
// @ts-ignore
export * from './matchStore';
// @ts-ignore
export * from './musicShown';
// @ts-ignore
export * from './nextRound';
// @ts-ignore
export * from './nextRoundStartTime';
// @ts-ignore
export * from './nowPlaying';
// @ts-ignore
export * from './nowPlayingSource';
// @ts-ignore
export * from './obsConfig';
// @ts-ignore
export * from './obsCredentials';
// @ts-ignore
export * from './obsState';
// @ts-ignore
export * from './predictionStore';
// @ts-ignore
export * from './radiaSettings';
// @ts-ignore
export * from './roundStore';
// @ts-ignore
export * from './runtimeConfig';
// @ts-ignore
export * from './scoreboardData';
// @ts-ignore
export * from './swapColorsInternally';
// @ts-ignore
export * from './tournamentData';
// @ts-ignore
export * from './bundleDeclaredConfig';
// @ts-ignore
export * from './bundleCasterSets';
