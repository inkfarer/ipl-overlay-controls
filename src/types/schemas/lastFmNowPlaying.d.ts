/* prettier-ignore */
/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export type LastFmNowPlaying = SongInfo & {
	album?: string;
	cover?: string;
	artistSong?: string;
	[k: string]: unknown;
};

export interface SongInfo {
	artist?: string;
	song: string;
	[k: string]: unknown;
}
