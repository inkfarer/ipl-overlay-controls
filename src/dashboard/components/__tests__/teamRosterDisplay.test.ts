import TeamRosterDisplay from '../teamRosterDisplay.vue';
import { mount } from '@vue/test-utils';
import { ActiveRoundTeam } from 'schemas';

describe('TeamRosterDisplay', () => {
    it('matches snapshot', () => {
        const wrapper = mount(TeamRosterDisplay, {
            props: {
                team: {
                    name: 'Cool Team',
                    players: [
                        { name: 'player one' },
                        { name: 'player two' },
                        { name: 'player three' },
                        { name: 'player four' }
                    ]
                } as ActiveRoundTeam
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot with overlong names', () => {
        const wrapper = mount(TeamRosterDisplay, {
            props: {
                team: {
                    name: 'Cool Team Cool Team Cool Team Cool Team Cool Team Cool Team',
                    players: [
                        { name: 'player one' },
                        { name: 'player two' },
                        { name: 'player three player three player three player three player three player three player three player three' },
                        { name: 'player four' }
                    ]
                } as ActiveRoundTeam
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
