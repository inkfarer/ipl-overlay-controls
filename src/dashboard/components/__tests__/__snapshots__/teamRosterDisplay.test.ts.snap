// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamRosterDisplay matches snapshot 1`] = `
"<div>
  <div class="text-bold m-b-4 wrap-anywhere text-size-medium">Cool Team</div>
  <div class="m-t-2 wrap-anywhere">player one</div>
  <div class="m-t-2 wrap-anywhere">player two</div>
  <div class="m-t-2 wrap-anywhere">player three</div>
  <div class="m-t-2 wrap-anywhere">player four</div>
</div>"
`;

exports[`TeamRosterDisplay matches snapshot with overlong names 1`] = `
"<div>
  <div class="text-bold m-b-4 wrap-anywhere text-size-medium">Cool Team Cool Team Cool Team Cool Team Cool ...</div>
  <div class="m-t-2 wrap-anywhere">player one</div>
  <div class="m-t-2 wrap-anywhere">player two</div>
  <div class="m-t-2 wrap-anywhere">player three player three player three player...</div>
  <div class="m-t-2 wrap-anywhere">player four</div>
</div>"
`;
